services/src/unit-test/java/uk/gov/moj/sdt/services/utils/MessagingUtilityTest.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.moj.sdt.services.utils;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.moj.sdt.domain.api.IBulkSubmission;
+import uk.gov.moj.sdt.domain.api.IIndividualRequest;
+import uk.gov.moj.sdt.domain.api.ITargetApplication;
+import uk.gov.moj.sdt.services.messaging.MessageWriter;
+import uk.gov.moj.sdt.services.messaging.SdtMessage;
+import uk.gov.moj.sdt.services.messaging.api.IMessageWriter;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+import uk.gov.moj.sdt.utils.transaction.synchronizer.MessageSynchronizer;
+import uk.gov.moj.sdt.utils.transaction.synchronizer.api.IMessageSynchronizer;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class MessagingUtilityTest extends AbstractSdtUnitTestBase {
+
+    @BeforeEach
+    @Override
+    public void setUp() {
+    }
+
+    @Test
+    public void testEnqueueRequestNew() {
+        // Given
+        IIndividualRequest individualRequest = mock(IIndividualRequest.class);
+        IBulkSubmission bulkSubmission = mock(IBulkSubmission.class);
+        ITargetApplication targetApplication = mock(ITargetApplication.class);
+        IMessageSynchronizer messageSynchronizer = mock(IMessageSynchronizer.class);
+        IMessageWriter messageWriter = mock(IMessageWriter.class);
+
+        when(individualRequest.getBulkSubmission()).thenReturn(bulkSubmission);
+        when(bulkSubmission.getTargetApplication()).thenReturn(targetApplication);
+        when(targetApplication.getTargetApplicationCode()).thenReturn("targetAppCode");
+        when(individualRequest.getSdtRequestReference()).thenReturn("sdtRequestReference");
+
+        MessagingUtility messagingUtility = new MessagingUtility(messageWriter,messageSynchronizer);
+        messagingUtility.setMessageSynchronizer(messageSynchronizer);
+        messagingUtility.setMessageWriter(messageWriter);
```

Comment:
See previous comment about making this a class field.

Additionally, best to just use the constructor here rather than both it and the set methods.  If you want coverage for the set methods it'll be better to do them in separate tests.

Reply:
done

Reply:
Remove the setMessageSynchronizer() and setMessageWriter() methods.  The constructor is already setting them and the two set methods are being explicitly tested in the following two tests.

Reply:
already removed